lib/plugins/prometheus/PrometheusCounterTransactionManager.ts

```diff
@@ -37,11 +45,38 @@ export class PrometheusCounterTransactionManager implements TransactionObservabi
     const transactionName = this.transactionNameByKey.get(uniqueTransactionKey)
     if (!transactionName) return
 
-    this.counter?.inc({ status: wasSuccessful ? 'success' : 'failed', transactionName })
+    const labels: Record<'status' | 'transactionName', string> = {
+      status: wasSuccessful ? 'success' : 'failed',
+      transactionName,
+    }
+    let labelsWithCustom: Record<'status' | 'transactionName' | CustomLabels, string> | undefined
+    if (this.customLabelsByKey.has(uniqueTransactionKey)) {
+      const customLabels = this.customLabelsByKey.get(uniqueTransactionKey)
+      labelsWithCustom = {
+        ...labels,
+        // biome-ignore lint/style/noNonNullAssertion: we already checked the presence
+        ...customLabels!,
+      }
+    }
+
+    this.counter?.inc(labelsWithCustom ?? labels)
     this.transactionNameByKey.delete(uniqueTransactionKey)
+    this.customLabelsByKey.delete(uniqueTransactionKey)
+  }
+
+  // Prometheus labels are the way Prometheus handles custom attributes
+  addCustomAttributes(
+    uniqueTransactionKey: string,
+    atts: { [p: string]: string | number | boolean },
+  ): void {
+    const transactionName = this.transactionNameByKey.get(uniqueTransactionKey)
+    if (!transactionName) return
+
+    // @ts-expect-error We only enforce types lightly here. If this ever causes us problem, we can start doing runtime validation here, but it seems to be an overkill for now.
```

Comment:
maybe we also should extend the interface of TransactionObservabilityManager to have a generic for customAttribute keys, but that could be a future improvement